lib/src/translator/translator.dart

```diff
@@ -84,4 +109,55 @@ class Translator {
     return translated;
   }
 
+  // translate matrix expression such as 'Matrix1+Matrix2'
+  List<String> _sanitizeMatrixInput(String input){
+    input = input.replaceAll("\$+&", "\$ + &");
+    input = input.replaceAll("\$-&", "\$ - &");
+    input = input.replaceAll("\$*&", "\$ * &");
+    input = input.replaceAll("\$/&", "\$ / &");
+
+    List<String> sanitizedInput = input.split(TranslatePattern.spacing).where((item) => item.isNotEmpty).toList();
+
+    return sanitizedInput;
+  }
+
+  // returns the Matrix size as a list of [row, col]
+  List<int> _matrixSize(String sanitizedInput){
+    List<String> matrixRow = sanitizedInput.replaceAll(RegExp(r'(&|\$)'), "").split("@").where((item) => item.isNotEmpty).toList();
+    int matrixRowSize = matrixRow.length;
+    int matrixColSize = matrixRow[0].split(";").where((item) => item.isNotEmpty).toList().length;
+    List<int> matrixSize = [matrixRowSize, matrixColSize];
+
+    return matrixSize;
+  }
+
+  String evaluateMatrix(List<int> matrixSize, List<String> matrixValues){
+    String matrix = "&";
+    int count = 0;
+
+    for(int r = 0; r < matrixSize[0]; r++){
+      for(int c = 0; c < matrixSize[1]; c++) {
+        String token = matrixValues[count];
+        // check if math expression
+        if (validateMatrix.isMathFunction(token)) {
+          // evaluate the math expression
+          String expression = translate(token, options);
+          double results = calculateFunction(Utf8.toUtf8(expression));  // call to backend evaluator
+          matrix += transformer.transform(results, options.decimalPlaces) + ";";
```

Comment:
Decimals shouldn't be transformed before the call, the calculation should happen and the result should be transformed on the way back. This should be handled in MatrixCalculator